packages/spacecat-shared-brand-client/src/index.js

```diff
@@ -0,0 +1,167 @@
+ */
+/* c8 ignore start */
+import { hasText, isValidUrl } from '@adobe/spacecat-shared-utils';
+import { ImsClient } from '@adobe/spacecat-shared-ims-client';
+import { context as h2, h1 } from '@adobe/fetch';
+
+export const { fetch } = process.env.HELIX_FETCH_FORCE_HTTP1
+  ? h1()
+  : h2();
+
+const PUBLISHED_BRANDS_FILTER = 'roles=BRAND&itemFilter=publishedBrands';
+const API_GET_BRANDS = `/api/v1/libraries?${PUBLISHED_BRANDS_FILTER}`;
+const API_GET_BRAND_GUIDELINES = (brandId) => `/api/v1/libraries/${brandId}?selector=details`;
+
+export default class BrandClient {
+  static createFrom(context) {
+    const { env, log = console } = context;
+    const { BRAND_API_BASE_URL: apiBaseUrl, BRAND_API_KEY: apiKey } = env;
+    log.info(`Creating BrandClient with apiBaseUrl: ${apiBaseUrl} and apiKey: ${apiKey}`);
+
+    if (context.brandClient) return context.brandClient;
+
+    const client = new BrandClient({ apiBaseUrl, apiKey }, log);
+    context.brandClient = client;
+    return client;
+  }
+
+  constructor({ apiBaseUrl, apiKey }, log) {
+    if (!isValidUrl(apiBaseUrl)) {
+      throw new Error(`Invalid Brand API Base URL: ${apiBaseUrl}`);
+    }
+    if (!hasText(apiKey)) {
+      throw new Error(`Invalid Brand API Key: ${apiKey}`);
+    }
+    this.apiBaseUrl = apiBaseUrl;
+    this.apiKey = apiKey;
+    this.log = log;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  mapToBrand(library) {
+    let createdAt = '';
+    let updatedAt = '';
+    try {
+      createdAt = new Date(library.created_date)?.toISOString();
+    } catch (e) {
+      this.log.error(`Error converting createdAt for brand ${library.library_urn}: ${e.message}`);
+    }
+    try {
+      updatedAt = new Date(library.modified_date)?.toISOString();
+    } catch (e) {
+      this.log.error(`Error converting updatedAt for brand ${library.library_urn}: ${e.message}`);
+    }
+    return {
+      id: library.library_urn,
+      name: library.name,
+      imsOrgId: library.org_id,
+      createdAt,
+      updatedAt,
+    };
+  }
+
+  async getBrandsForOrganization(imsOrgId, imsAccessToken) {
+    if (!hasText(imsOrgId)) {
+      throw new Error(`Invalid IMS Org ID: ${imsOrgId}`);
+    }
+
+    if (!hasText(imsAccessToken)) {
+      throw new Error(`Invalid IMS Access Token: ${imsAccessToken}`);
+    }
+    const headers = {
+      'Content-Type': 'application/json',
+      Authorization: `${imsAccessToken}`,
+      'x-api-key': this.apiKey,
+    };
+    const response = await fetch(`${this.apiBaseUrl}${API_GET_BRANDS}`, {
+      headers,
+    });
+
+    if (!response.ok) {
+      throw new Error(`Error getting brands for organization ${imsOrgId}: ${response.status}`);
+    }
+    try {
+      const result = await response.json();
+      return result.libraries.filter((library) => library.org_id === imsOrgId).map(this.mapToBrand);
+    } catch (e) {
+      this.log.error(`Error getting brands for organization: ${e.message}`);
```

Comment:
Let's add orgId to the log message, so we can monitor on spacecat side if smth doesn't work